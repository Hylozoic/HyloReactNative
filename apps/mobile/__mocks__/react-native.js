import React, {useState} from 'react'
import * as ReactNative from 'react-native'

export const Switch = props => {
  const [value, setValue] = useState(props.value)
  return (
    <ReactNative.TouchableOpacity
      onPress={() => {
        props.onValueChange(!value)
        setValue(!value)
      }}
      testID={props.testID}>
      <ReactNative.Text>{value?.toString()}</ReactNative.Text>
    </ReactNative.TouchableOpacity>
  )
}

newRN = Object.defineProperty(ReactNative, 'Switch', {
  get: function() {
    return Switch
  }
})

module.exports = newRN
