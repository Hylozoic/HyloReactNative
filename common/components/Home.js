import React from 'react'
import WelcomeScene from './WelcomeScene'
import MyPosts from './MyPosts'
import {
  Navigator,
  Text,
  TouchableOpacity
} from 'react-native'
import mixins from '../style/mixins'

const router = (route, navigator) => {
  switch (route.name) {
    case 'root':
      return <WelcomeScene
        showMyPosts={() => {
          navigator.push({
            title: 'Your Posts',
            name: 'myPosts',
            index: route.index + 1
          })
        }} />
    case 'myPosts':
      return <MyPosts title={route.title} />
  }
}

export default class Home extends React.Component {
  render () {
    const routeMapper = {
      LeftButton: (route, navigator, index, navState) => {
        if (route.index === 0) {
          return null
        }
        return <TouchableOpacity onPress={() => navigator.pop()}>
          <Text style={styles.navigationLeftButton}>&lt; Back</Text>
        </TouchableOpacity>
      },
      RightButton: (route, navigator, index, navState) => {
        return null
      },
      Title: (route, navigator, index, navState) => {
        return <Text style={styles.navigationTitle}>{route.title}</Text>
      }
    }

    return <Navigator
      initialRoute={{name: 'root', title: 'Welcome', index: 0}}
      renderScene={router}
      navigationBar={<Navigator.NavigationBar
        style={styles.navigationBar}
        routeMapper={routeMapper} />} />
  }
}

const styles = {
  navigationBar: {
    backgroundColor: '#22bf99'
  },
  navigationTitle: {
    ...mixins.navigationText
  },
  navigationLeftButton: {
    ...mixins.navigationText,
    paddingLeft: 10,
    paddingRight: 10
  }
}
