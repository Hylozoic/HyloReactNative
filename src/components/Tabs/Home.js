import React from 'react'

import Feed from '../Feed'
import header from './header'

const title = 'Home'

export default class Home extends React.Component {
  static navigationOptions = ({navigation}) => (header(navigation, title))
  render () {
    return <Feed navigation={this.props.navigation} />
  }
}
