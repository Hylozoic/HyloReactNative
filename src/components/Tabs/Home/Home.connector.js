import { connect } from 'react-redux'
import { get } from 'lodash/fp'

import getMe from '../../../store/selectors/getMe'
import getCurrentCommunityId from '../../../store/selectors/getCurrentCommunityId'
import getCurrentNetworkId from '../../../store/selectors/getCurrentNetworkId'
import { mapWhenFocused } from 'navigation/util/redux'

export function mapStateToProps (state, props) {
  return {
    currentUser: getMe(state, props),
    communityId: getCurrentCommunityId(state, props),
    networkId: getCurrentNetworkId(state, props)
  }
}

export default connect(mapWhenFocused(mapStateToProps))
