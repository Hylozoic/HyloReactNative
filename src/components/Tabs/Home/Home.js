import React from 'react'

import Feed from '../../Feed'
import Header from '../Header'
import Loading from '../../Loading'

export default class Home extends React.Component {
  static navigationOptions = ({ navigation, screenProps }) => {
    return Header(navigation, screenProps.currentTabName)
  }

  componentDidMount () {
    this.props.navigation.navigate('PostEditor', {communityId: '9'})
  }

  shouldComponentUpdate (nextProps) {
    return nextProps.isFocused
  }

  render () {
    const { communityId, currentUser, navigation, networkId } = this.props
    if (!currentUser) return <Loading style={{flex: 1}} />
    return <Feed
      communityId={communityId}
      navigation={navigation}
      networkId={networkId}
      screenProps={this.props.screenProps} />
  }
}
