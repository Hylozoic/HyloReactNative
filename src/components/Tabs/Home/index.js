import { withNavigationFocus } from 'react-navigation-is-focused-hoc'

import component from './Home'
import connector from './Home.connector'

export default connector(withNavigationFocus(component))
