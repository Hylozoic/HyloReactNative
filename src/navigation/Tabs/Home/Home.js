import React from 'react'
import Feed from 'navigation/Feed'
import Loading from 'components/Loading'

export default class Home extends React.Component {
  shouldComponentUpdate (nextProps) {
    return nextProps.isFocused
  }

  render () {
    const { communityId, currentUser, route, navigation, networkId } = this.props
    if (!currentUser) return <Loading style={{ flex: 1 }} />

    return (
      <Feed
        communityId={communityId}
        route={route}
        navigation={navigation}
        networkId={networkId}
      />
    )
  }
}
