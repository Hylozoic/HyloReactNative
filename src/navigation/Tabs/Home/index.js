import { withNavigationFocus } from '@react-navigation/compat'

import component from './Home'
import connector from './Home.connector'

export default withNavigationFocus(connector(component))
