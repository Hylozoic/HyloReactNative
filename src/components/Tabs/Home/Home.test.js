import 'react-native'
import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { Home } from './Home'

jest.mock('react-native-device-info')

it('shows Loading with no community or user', () => {
  const renderer = new ReactShallowRenderer()
  renderer.render(<Home />)
  const actual = renderer.getRenderOutput()

  expect(actual).toMatchSnapshot()
})

it('shows Feed with a community and user', () => {
  const renderer = new ReactShallowRenderer()
  renderer.render(<Home currentUser={{}} communityId='1' />)
  const actual = renderer.getRenderOutput()

  expect(actual).toMatchSnapshot()
})
