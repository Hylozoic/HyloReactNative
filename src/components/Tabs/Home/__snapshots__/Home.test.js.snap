// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows Feed with a community and user 1`] = `
<Connect(Feed)
  communityId="1"
  navigation={undefined}
  networkId={undefined}
  screenProps={undefined}
/>
`;

exports[`shows Loading with no community or user 1`] = `<Loading />`;
